Add findsplinemax(), refactor minspline.py into minmaxspline.py

brews opened the pull request:
Changes in this PR:
- Adds `impactcommon.math.minmaxspline.findsplinemax()`. It finds the local maxima of an open estimate-style cubic spline `Curve`.
- Refactors logic for `findsplinemin()` to `impactcommon/math/minmaxspline.py` but maintains reference in `impactcommon/math/minspline.py` for backwards compatibility.
- Adds unit tests for new `findsplinemax()`.

Needed for downstream labor sector projection development.

Close #15

brews committed:
Add unittest for minmaxspline

brews committed:
Add minmaxspline submodule, includes findsplinemin() and findsplinemax()

Close #15

brews committed:
Refactors minspline.py to pull from minmaxspline.py for backwards compatibility

jrising reviewed (commented):
I don't love this solution. You can get the maxima by just passing in negative coefficients, and instead you've made it so that one always needs to compute both. And we will basically never want both. Why not just create `findsplinemax` which calls `findsplinemin` with the negative coefficients?

brews committed:
Follow reviewer feedback, findsplinemax is findsplinemin but -1 * coeffs

This redesign is useful because we don't expect to ever need the spline min
and max together. Most usecases are for minima or the maxima, not both.

brews commented:
A fine idea, @jrising . Fixed in 6988966.

jrising reviewed (approved):
Great. Thanks for this.
